sdk/core/core-rest-pipeline/src/policies/proxyPolicy.ts

```diff
@@ -155,7 +158,8 @@ export function getProxyAgentOptions(
     hostname: parsedProxyUrl.hostname,
     port: proxySettings.port,
     protocol: parsedProxyUrl.protocol,
-    headers: requestHeaders.toJSON(),
+    headers: headers.toJSON(),
+    ...tlsSettings,
```

Comment:
Did you try this with a live service? I wonder how proxy works when the service requires a custom CA/cert to authenticate.

Reply:
I haven't. I don't know a service that is configured this way. I'll check to see if I come up with a way to test this scenario

Reply:
I guess my question is if we take a service that uses CA certificate authentication (I think service-fabric-rest does this?) and we use some local TLS-enabled proxy, can we get requests to go through?

Let me know if the scenario doesn't make sense, I'm not a proxy expert.